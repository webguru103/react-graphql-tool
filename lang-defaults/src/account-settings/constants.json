[
  {
    "id": "first-name-too-long",
    "defaultMessage": "First name is too long, must be less than or equal to 40 characters"
  },
  {
    "id": "last-name-too-long",
    "defaultMessage": "Last name is too long, must be less than or equal to 40 characters"
  },
  {
    "id": "name-too-long",
    "defaultMessage": "Name is too long, first and last name must be each less than 40 characters"
  }
]