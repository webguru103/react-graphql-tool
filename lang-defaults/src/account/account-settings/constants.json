[
  {
    "id": "password-not-changed",
    "defaultMessage": "The old and new passwords are the same."
  }
]