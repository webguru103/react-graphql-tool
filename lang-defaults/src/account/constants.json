[
  {
    "id": "sign-up",
    "defaultMessage": "Sign Up"
  },
  {
    "id": "sign-up-as",
    "defaultMessage": "Sign up as"
  },
  {
    "id": "invited-to-sign-up",
    "defaultMessage": "You have been invited to sign up as:"
  },
  {
    "id": "activating-account",
    "defaultMessage": "Activating your account"
  },
  {
    "id": "account-activated",
    "defaultMessage": "Account activated"
  },
  {
    "id": "could-not-activate",
    "defaultMessage": "Could not activate account"
  },
  {
    "id": "please-wait",
    "defaultMessage": "Please wait..."
  },
  {
    "id": "successfully-activated",
    "defaultMessage": "Your account has been successfully activated. You can now log in."
  },
  {
    "id": "email-already-registered",
    "defaultMessage": "You already have a DealTap account. Confirm your password to continue."
  },
  {
    "id": "email-already-taken",
    "defaultMessage": "This email is already taken. Please try another one."
  },
  {
    "id": "empty-email",
    "defaultMessage": "The email is empty"
  },
  {
    "id": "invalid-email",
    "defaultMessage": "The email is invalid"
  },
  {
    "id": "network-error",
    "defaultMessage": "Something went wrong, please try again later..."
  },
  {
    "id": "wrong-username-password",
    "defaultMessage": "Email or password is wrong"
  },
  {
    "id": "email-empty",
    "defaultMessage": "Please input your email address"
  },
  {
    "id": "email-doesnt-exist",
    "defaultMessage": "Email address does not exist"
  },
  {
    "id": "email-resent",
    "defaultMessage": "The email has been sent again. Please check your email"
  },
  {
    "id": "request-error",
    "defaultMessage": "Sorry, there was an error processing your request!"
  },
  {
    "id": "request-being-processed",
    "defaultMessage": "Your request is being processed."
  },
  {
    "id": "brokerage-accepted",
    "defaultMessage": "Invitation accepted"
  },
  {
    "id": "could-not-accept-brokerage",
    "defaultMessage": "Invitation not accepted"
  },
  {
    "id": "brokerage-rejected",
    "defaultMessage": "Invitation rejected"
  },
  {
    "id": "could-not-reject-brokerage",
    "defaultMessage": "Invitation not rejected"
  },
  {
    "id": "could-not-find-invite",
    "defaultMessage": "No invite found."
  },
  {
    "id": "link-expired",
    "defaultMessage": "It appears this link has already expired."
  },
  {
    "id": "joined-brokerage",
    "defaultMessage": "Joined Brokerage Office"
  },
  {
    "id": "failed-to-join-brokerage",
    "defaultMessage": "Failed To Join Brokerage Office"
  },
  {
    "id": "already-joined-brokerage",
    "defaultMessage": "Already Joined a Brokerage Office"
  },
  {
    "id": "already-joined-brokerage-message",
    "defaultMessage": "It appears you have already joined a brokerage office as an agent."
  },
  {
    "id": "already-granted-permission",
    "defaultMessage": "Already granted permission"
  },
  {
    "id": "already-have-access",
    "defaultMessage": "Already Have Access"
  },
  {
    "id": "already-have-access-message",
    "defaultMessage": "It appears you already have access to Control Panel. Please log in instead."
  },
  {
    "id": "permission-granted",
    "defaultMessage": "Permission Granted"
  },
  {
    "id": "failed-to-grant-permission",
    "defaultMessage": "Failed to Grant Permission"
  },
  {
    "id": "get-agent-panel-access",
    "defaultMessage": "Get Agent Panel Access"
  },
  {
    "id": "get-agent-panel-access-message",
    "defaultMessage": "We already have your information since you have a DealTap account. Click confirm to get agent panel access."
  },
  {
    "id": "agent-identity-added",
    "defaultMessage": "Agent Identity Added"
  },
  {
    "id": "agent-identity-added-message",
    "defaultMessage": "You can now access DealTap's agent panel."
  },
  {
    "id": "account-created",
    "defaultMessage": "Account Created"
  },
  {
    "id": "account-created-message",
    "defaultMessage": "Your account has been created."
  },
  {
    "id": "failed-to-reject",
    "defaultMessage": "Failed to reset password"
  },
  {
    "id": "password-reset-success-title",
    "defaultMessage": "Password has been reset"
  },
  {
    "id": "password-reset-success-message",
    "defaultMessage": "Please log in to continue."
  },
  {
    "id": "reset-password",
    "defaultMessage": "Reset Password"
  }
]