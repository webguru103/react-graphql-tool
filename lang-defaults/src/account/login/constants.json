[
  {
    "id": "account-not-verified",
    "defaultMessage": "We noticed you still haven't verified your email address. We have sent another verification email. Please check your inbox."
  },
  {
    "id": "user-not-exists",
    "defaultMessage": "Sorry, we couldnâ€™t recognize this email"
  },
  {
    "id": "email-not-valid",
    "defaultMessage": "Please input a valid email address."
  },
  {
    "id": "email-empty",
    "defaultMessage": "Please input your email address"
  },
  {
    "id": "verify-email",
    "defaultMessage": "Verify your email"
  },
  {
    "id": "network-message",
    "defaultMessage": "Some error occurred, try again later"
  }
]