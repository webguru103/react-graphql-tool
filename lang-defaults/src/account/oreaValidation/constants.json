[
  {
    "id": "cannot-reach-title",
    "defaultMessage": "We couldn't reach OREA"
  },
  {
    "id": "make-request-msg",
    "defaultMessage": "In order to continue,\n    we need to verify your identity with OREA.\n    Please click the button below to verify\n    your identity."
  },
  {
    "id": "first-login-msg-accent",
    "defaultMessage": "In order to use OREA forms,"
  },
  {
    "id": "first-login-msg",
    "defaultMessage": "you need to verify your identity with OREA.\n    Please click the button below to verify\n    your identity."
  },
  {
    "id": "make-request-title",
    "defaultMessage": "Identity verification"
  },
  {
    "id": "cannot-reach-msg",
    "defaultMessage": "Sorry, but we were not able to reach\n    OREA at this moment. You can go back and\n    try again, or verify later in your account settings."
  },
  {
    "id": "do-it-later",
    "defaultMessage": "Do It Later"
  },
  {
    "id": "try-again",
    "defaultMessage": "Try Again"
  },
  {
    "id": "do-it-later-msg",
    "defaultMessage": "You can validate with OREA anytime in the Account Settings page."
  },
  {
    "id": "ok-button",
    "defaultMessage": "OK"
  },
  {
    "id": "cancel-button",
    "defaultMessage": "Cancel"
  },
  {
    "id": "do-it-later-button",
    "defaultMessage": "I'll do it later."
  },
  {
    "id": "go-to-orea",
    "defaultMessage": "Go to OREA validation page"
  },
  {
    "id": "ver-failure-msg",
    "defaultMessage": "Sorry, but we were not able to verify your identity with OREA. Please check your information and try again."
  },
  {
    "id": "ver-failure-title",
    "defaultMessage": "Verification failed"
  },
  {
    "id": "ver-sucess-msg",
    "defaultMessage": "Your identity has been successfully verified through OREA."
  },
  {
    "id": "ver-sucess-title",
    "defaultMessage": "OREA verificiation successful"
  },
  {
    "id": "done-button",
    "defaultMessage": "Done"
  }
]