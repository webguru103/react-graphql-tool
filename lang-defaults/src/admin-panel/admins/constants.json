[
  {
    "id": "request-error",
    "defaultMessage": "Sorry, there was an error processing your request!"
  },
  {
    "id": "success-invite",
    "defaultMessage": "The admin has been successfully invited"
  }
]