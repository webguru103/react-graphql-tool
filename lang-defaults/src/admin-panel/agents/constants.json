[
  {
    "id": "successfuly-sent-invite",
    "defaultMessage": "Invite has been sent"
  },
  {
    "id": "successfuly-sent-invites",
    "defaultMessage": "Successfuly sent {amount} invites"
  },
  {
    "id": "no-emails",
    "defaultMessage": "Please input an email"
  },
  {
    "id": "invalid-emails",
    "defaultMessage": "Please input valid emails separated with comma or return"
  },
  {
    "id": "request-error",
    "defaultMessage": "Sorry, there was an error processing your request!"
  },
  {
    "id": "couldnot-send-invite",
    "defaultMessage": "Could not send invite to the following agents:"
  }
]