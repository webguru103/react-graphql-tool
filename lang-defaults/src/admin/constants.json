[
  {
    "id": "empty-email",
    "defaultMessage": "Please input an email"
  },
  {
    "id": "empty-first-name",
    "defaultMessage": "Please input a first name"
  },
  {
    "id": "empty-last-name",
    "defaultMessage": "Please input a last name"
  },
  {
    "id": "request-error",
    "defaultMessage": "Sorry, there was an error processing your request!"
  }
]