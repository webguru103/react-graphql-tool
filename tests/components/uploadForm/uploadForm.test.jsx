/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import 'jest-dom/extend-expect';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { withContext } from '../../test-helpers/contextWrapper';
import UploadForm from '../../../src/shared/uploadForm/uploadForm';
import { MAX_UPLOAD_SIZE } from '../../../src/control-panel/form-management/constants';

describe('UploadForm', () => {
  afterEach(cleanup);
  const Component = withContext(UploadForm);

  test('renders', () => {
    const { container } = render(<Component
      xhr={new XMLHttpRequest()}
      acceptTypes={['application/pdf']}
      acceptMultiple={false}
      maxUploadSize={MAX_UPLOAD_SIZE}
      uploadLabel={<FormattedMessage id="upload-label" defaultMessage="Upload PDF, JPG, PNG" />}
      onDrop={() => {}}
      setUploadError={() => {}}
      setURLs={() => {}}
      uploadEndpoint=""
    />);
    expect(container).toMatchSnapshot();
  });

  test('progress bar and cancel button should display when upload in progress', () => {
    const xhr = new XMLHttpRequest();
    const { getByTestId } = render(<Component
      xhr={xhr}
      acceptTypes={['application/pdf']}
      acceptMultiple={false}
      maxUploadSize={MAX_UPLOAD_SIZE}
      uploadLabel={<FormattedMessage id="upload-label" defaultMessage="Upload PDF, JPG, PNG" />}
      onDrop={() => {}}
      setUploadError={() => {}}
      setURLs={() => {}}
      uploadEndpoint=""
      testMode
    />);

    const progressEvent = new ProgressEvent('progress', {
      lengthComputable: true,
      loaded: 500,
      total: 1000,
    });
    fireEvent(xhr.upload, progressEvent);

    // Progress bar should be present
    getByTestId('linear-progress');
    getByTestId('cancel-button');
  });

  test('progress bar & cancel button should be hidden if progress is zero', () => {
    const xhr = new XMLHttpRequest();
    const { getByTestId } = render(<Component
      xhr={xhr}
      acceptTypes={['application/pdf']}
      acceptMultiple={false}
      maxUploadSize={MAX_UPLOAD_SIZE}
      uploadLabel={<FormattedMessage id="upload-label" defaultMessage="Upload PDF, JPG, PNG" />}
      onDrop={() => {}}
      setUploadError={() => {}}
      setURLs={() => {}}
      uploadEndpoint=""
      testMode
    />);

    // Should not throw error
    expect(() => getByTestId('linear-progress')).toThrow();
    expect(() => getByTestId('cancel-button')).toThrow();
  });

  test('should render a processing button once progress=100% and should reset network error on click', () => {
    const handleResetNetworkErrorOnCancel = jest.fn();

    const xhr = new XMLHttpRequest();
    const { getByTestId } = render(<Component
      xhr={xhr}
      acceptTypes={['application/pdf']}
      acceptMultiple={false}
      maxUploadSize={MAX_UPLOAD_SIZE}
      uploadLabel={<FormattedMessage id="upload-label" defaultMessage="Upload PDF, JPG, PNG" />}
      onDrop={() => {}}
      setUploadError={handleResetNetworkErrorOnCancel}
      setURLs={() => {}}
      uploadEndpoint=""
      testMode
    />);

    const progressEvent = new ProgressEvent('progress', {
      lengthComputable: true,
      loaded: 1000,
      total: 1000,
    });
    fireEvent(xhr.upload, progressEvent);

    const processingButton = getByTestId('processing');

    fireEvent(
      processingButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );

    expect(handleResetNetworkErrorOnCancel).toHaveBeenCalledTimes(1);

  });
});
