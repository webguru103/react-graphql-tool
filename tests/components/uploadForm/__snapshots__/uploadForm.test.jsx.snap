// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadForm renders 1`] = `
<div>
  <div
    class="UploadFormC-dropzoneContainer-6"
    data-testid="uploadForm"
  >
    <div
      aria-disabled="false"
      class="Dropzone-root-20"
      data-testid="dropzone"
      style="position: relative;"
    >
      <div
        class="UploadFormC-dropzone-1 UploadFormC-paddingAnimationDropzoneOff-8"
      >
        <svg
          fill="none"
          height="34"
          viewBox="0 0 37 34"
          width="37"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M27.75 9.25c-.077 0-.154 0-.23.003C26.445 3.954 21.76 0 16.187 0 9.802 0 4.624 5.177 4.624 11.563c0 .462.027.922.081 1.377a7.708 7.708 0 0 0 3.002 14.81 1.542 1.542 0 1 0 0-3.083 4.625 4.625 0 0 1-.934-9.156 1.542 1.542 0 0 0 1.187-1.884 8.48 8.48 0 0 1 8.227-10.543c4.49 0 8.194 3.5 8.463 7.959.057.936.93 1.606 1.85 1.417a6.167 6.167 0 1 1 1.25 12.207 1.542 1.542 0 1 0-.001 3.083 9.25 9.25 0 0 0 0-18.5zm-8.48 22.958V20.513L21 22.242a1.542 1.542 0 1 0 2.18-2.18l-3.27-3.27a3.083 3.083 0 0 0-4.361 0l-3.27 3.27a1.542 1.542 0 0 0 2.18 2.18l1.729-1.729v11.695a1.542 1.542 0 0 0 3.083 0z"
            fill="#7A49DF"
            fill-rule="evenodd"
          />
        </svg>
        <span>
          Upload PDF, JPG, PNG
        </span>
      </div>
      <input
        accept="application/pdf"
        autocomplete="off"
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px; opacity: 0.00001; pointer-events: none;"
        type="file"
      />
    </div>
    
  </div>
</div>
`;
